aleph-client/src/contract/convertible_value.rs

```diff
@@ -193,3 +186,124 @@ where
         }
     }
 }
+
+impl<Elem: TryFrom<ConvertibleValue, Error = anyhow::Error>> TryFrom<ConvertibleValue>
+    for Vec<Elem>
+{
+    type Error = anyhow::Error;
+
+    fn try_from(value: ConvertibleValue) -> std::result::Result<Self, Self::Error> {
+        let seq = match value.0 {
+            Value::Seq(seq) => seq,
+            _ =>  bail!("Failed parsing `ConvertibleValue` to `Vec<T>`. Expected `Seq(_)` but instead got: {:?}", value),
+        };
+
+        let mut result = vec![];
+        for element in seq.elems() {
+            result.push(ConvertibleValue(element.clone()).try_into()?);
+        }
+
+        Ok(result)
+    }
+}
+
+impl<const N: usize, Elem: TryFrom<ConvertibleValue, Error = anyhow::Error> + Debug>
+    TryFrom<ConvertibleValue> for [Elem; N]
+{
+    type Error = anyhow::Error;
+
+    fn try_from(value: ConvertibleValue) -> std::result::Result<Self, Self::Error> {
+        Vec::<Elem>::try_from(value)?
+            .try_into()
+            .map_err(|e| anyhow!("Failed to convert vector to an array: {e:?}"))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use contract_transcode::Value::{Bool, Char, Int, Seq, UInt};
+
+    use crate::contract::ConvertibleValue;
+
+    #[test]
+    fn converts_boolean() {
+        let cast: bool = ConvertibleValue(Bool(true))
+            .try_into()
+            .expect("Should cast successfully");
+        assert_eq!(true, cast);
```

Comment:
`assert2` would be nice, but then it should be applied across the whole crate and I'm not eager to do that in this PR :|

one-liners would be cool indeed, but there are two problems:
 - no implementation for `anyhow::Error == anyhow::Error`, so there must be unwrapping
 - the first line has type ascription, which is required while casting; otherwise I would have to use `assert_eq!(-41, <ConvertibleValue as TryInto<i32>>::try_into(ConvertibleValue(Int(-41))).unwrap());`, which is pretty ugly